### PR TITLE
Fix Social CTA on My Jetpack

### PR DESCRIPTION
**Pull request by IanRamosC**
## Proposed changes:

* This PR adds `has_required_plan` to Social product class. This method is responsible for checking whether a site has a plan for a specific product. Adding this method, Social now also shows the "Learn More" button if the site doesn't have Social, instead of just the "Activate" button.
image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pbNhbs-9bI-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Spin up a JN site with Jetpack Beta pointing to this branch
* Connected to Jetpack, visit the My Jetpack page
* If you don't have a Social/Complete plan. Confirm that the CTA for Social says "Learn More" instead of "Activate"

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-social.php [hunk 1]
```diff
@@ -127,6 +127,27 @@ public static function get_wpcom_product_slug() {
 		return 'jetpack_social';
```
Comment (jboland88) [inline projects/packages/my-jetpack/src/products/class-social.php]: I think we need to change this to an active product slug - maybe `jetpack_social_basic_yearly`? This is the slug the interstitial is going to the cart with

**Review by jboland88 — CHANGES_REQUESTED** [review]
The code looks good, but I get a cart error when clicking on the interstitial CTA. I think we just need to change the product slug in the social class.

Screenshot 2024-01-08 at 2 56 47 PM

File: projects/packages/my-jetpack/src/products/class-social.php (hunk 1 above)
Comment (IanRamosC) [reply projects/packages/my-jetpack/src/products/class-social.php]: Nice catch. Didn't get that one in my local env.

**Review by jboland88 — APPROVED** [review]
thanks for the update - tested as expected!

**Pull request merged**